### PR TITLE
Use unmodifiable list of interceptors for thread safety

### PR DESCRIPTION
**Pull request by jbarr21**
Fix `AssertionError` in `InterceptorChain` due to empty interceptor list when an `AsyncLayoutInflater` is used.

### CODE REVIEW

**Comment by chrisjenx** [comment]
Are you able to give a bit more information around what was happening and how this fixes it?

**Comment by jbarr21** [comment]
> Are you able to give a bit more information around what was happening and how this fixes it?
the ViewPump singleton object's `mInterceptorsWithFallback` list was getting cleared and having all interceptors plus the fallback being re-added back to the list on each request. With the introduction of an `AsyncLayoutInflater`, the `ViewPump` object can be accessed from multiple threads at the same time, and sometimes the list was empty if the list was cleared. by switching to a final immutable list, the interceptor list will not be modified, so it is safe to iterate across multiple threads

File: viewpump/src/main/java/io/github/inflationx/viewpump/ViewPump.java [hunk 1]
```diff
@@ -69,6 +64,11 @@ public boolean isCustomViewCreation() {
     public static Builder builder() {
         return new Builder();
     }
+    
+    /** Returns an immutable copy of {@code list}. */
+    private static <T> List<T> immutableList(List<T> list) {
+        return Collections.unmodifiableList(new ArrayList<>(list));
```
Comment (chrisjenx) [inline viewpump/src/main/java/io/github/inflationx/viewpump/ViewPump.java]: Does wrapping with the `unmodifiableList` not create a copy? Just wondering if you need to call the new ArrayList here?

Comment (jbarr21) [inline viewpump/src/main/java/io/github/inflationx/viewpump/ViewPump.java]: no, its just a wrapper that implements list

**Pull request merged**